src/connection.ts

```diff
@@ -1,8 +1,8 @@
-import HostipWebSocket from './websocket/host-ip-websocket';
+import HostipWebSocket from "./websocket/host-ip-websocket";
 
-export default class Connection
-{
-    hostname : string;
-    clientId : string;
-    websocket: HostipWebSocket;
-}
\ No newline at end of file
+export default class Connection {
+  hostname: string;
+  clientId: string;
+  websocket: HostipWebSocket;
+  sockets?: Map<string, HostipWebSocket>;
```

Comment:
Interesting, so the websocket connections get stored in this new map?

Reply:
Yep, so there are X remote websocket connections to the server, a single connection between client/server, and X local websocket connections. The string in this map identifies each remote websocket, this information is included in each message so it gets delivered to the correct local websocket.

One thing that doesn't work reliably is closed sockets: some remote errors cannot be replayed locally, error 1006 I think.